stdlib/public/core/Hasher.swift

```diff
@@ -90,7 +90,10 @@ internal struct _HasherTailBuffer {
   internal init(tail: UInt64, byteCount: UInt64) {
     // byteCount can be any value, but we only keep the lower 8 bits.  (The
     // lower three bits specify the count of bytes stored in this buffer.)
-    _sanityCheck(tail & ~(1 << ((byteCount & 7) << 3) - 1) == 0)
+    // FIXME: The "as UInt64" annotations here eliminate some exponential
+    // behavior in the expression type checker <rdar://problem/42672946>.
+    _sanityCheck(tail & ~((1 as UInt64) << ((byteCount & (7 as UInt64))
+          << (3 as UInt64)) - (1 as UInt64)) == (0 as UInt64))
```

Comment:
`as` still has to consider bridging as a possibility. Is there any more blood to be squeezed from this stone by assigning these to explicitly-typed constants instead?

```
let one: UInt64 = 1
let seven: UInt64 = 7
let three: UInt64 = 3
tail & ~(one << ...
```

Reply:
Maybe! I'm not sure it'll make enough difference to matter, but maybe it'll be less ugly.

Reply:
Another possibility after SE-0213 is to do `UInt64(1)` since that's going to be an ascription as well for literals.

Reply:
I found a formulation that's faster to type-check.